trap.go

```diff
@@ -248,16 +248,19 @@ func (t *TrapListener) listenUDP(addr string) error {
 				snmpEngineID := securityParams.AuthoritativeEngineID
 				msgAuthoritativeEngineID := packetSecurityParams.AuthoritativeEngineID
 				if msgAuthoritativeEngineID != snmpEngineID {
-					if 4 < len(msgAuthoritativeEngineID) && len(msgAuthoritativeEngineID) < 32 {
-						// RFC3414 3.2.3a: continue processing
-					} else {
+					if len(msgAuthoritativeEngineID) < 5 || len(msgAuthoritativeEngineID) > 32 {
```

Comment:
The condition has voluntarily been made less restrictive by allowing engineID of exactly 32 characters as suggested in RFC3414